tokio-util/src/codec/length_delimited.rs

```diff
@@ -386,6 +394,27 @@ use std::{cmp, fmt, mem};
 /// `Builder` enables constructing configured length delimited codecs. Note
 /// that not all configuration settings apply to both encoding and decoding. See
 /// the documentation for specific methods for more detail.
+///
+/// # Configuration Compatibility
+/// Not all combinations of [`max_frame_length`], [`length_field_length`], and
+/// [`length_adjustment`] is valid. For a combination of these values to be valid,
+/// the following must be true:
+/// ```text
+/// max_frame_length < 2^(length_field_length * 8) + length_adjustment
+/// ```
+/// For example in the simplest case where `length_adjustment` is `0` and `length_field_length`
+/// is one byte, the smallest number that cannot be represented by it is 256. So `max_frame_length`
+/// must be smaller than 256 for the configuration to be valid.
+///
+/// `length_adjustment` can offset the maximum allowed value. For example if `length_adjustment`
+/// is `-1`, this means the length of the payload is going to be reported `1` byte more than its
+/// true value. So a payload of `255` bytes will be framed with a length of `256`. But `256` cannot
+/// be stored in a single byte `length_field_length`. So the previously valid `max_frame_length` of
+/// `255` is now invalid. The case for a positive `length_adjustment` can be inferred accordingly.
```

Comment:
I don't think you need to explain in such detail. You can just say something along the lines of "A configuration is only valid if the maximum frame length is small enough to be represented by the chosen length field size."

Reply:
Yeah you're right. I'll shorten/simplify the explanation.